docs: new wallet client

JasonPaulGithub opened the pull request.

maqi committed:
fix(wallet): remove unconfirmed_spends file from disk when all confirmed

maqi committed:
test(CI): ensure no extra leftover cash_note or payment files

maqi committed:
fix(client): avoid deadlock during upload in case of error

Commit:
docs: new wallet client

jacderida commented on sn_client/src/wallet.rs:
```diff
@@ -35,6 +35,7 @@ pub struct WalletClient {
 
 impl WalletClient {
     /// Create a new wallet client.
```
Can this text here also go in the markdown document? Sorry, I've never used this feature before.

JasonPaulGithub replied:
Yes, if this is agreed upon it will be the same for all documentation.
I left it in the rs file because it gives a quick overview to the reader (leaving the md for more detail)

actions-user committed:
chore(release): sn_cli-v0.88.17/sn_client-v0.101.11/sn_faucet-v0.2.17/sn_node-v0.102.17/sn_node_rpc_client-v0.2.17/sn_testnet-v0.3.16

jacderida committed:
chore: remove arg and env variable combination

Our code was setup such that the `--peer` arguments and those defined using `SAFE_PEERS` were
combined together to form one list. Now we use `clap` to parse one or the other.

joshuef commented on sn_client/src/docs/wallet/new.md:
````diff
@@ -0,0 +1,10 @@
+# Example
+
+ ```ignore
+ let mut wallet_client = WalletClient::new(client.clone(), wallet);
````
`ignore` is not the same ` dont run`, I think ? (Are the docs compiled and verified?)

Here we're missing all the setup and related info that would make this useful, no? (the why as you said)

joshuef commented on sn_client/src/wallet.rs:
```diff
@@ -35,6 +35,7 @@ pub struct WalletClient {
 
 impl WalletClient {
     /// Create a new wallet client.
+    #[doc = include_str!("docs/wallet/new.md")]
```
I don't think there's any reason to move the docs to another file here. 

As I mentioned, there's an extra burden to move to update these docs, so to my mind unless they are _huge_ we'd want to keep them in the same file.

maqi committed:
fix(CI): set correct log level for bench test

actions-user committed:
chore(release): sn_cli-v0.88.18/sn_faucet-v0.2.18/sn_node-v0.102.18/sn_node_rpc_client-v0.2.18/sn_testnet-v0.3.17

actions-user committed:
chore(release): sn_cli-v0.88.19/sn_peers_acquisition-v0.2.2/sn_faucet-v0.2.19/sn_node-v0.102.19/sn_node_rpc_client-v0.2.19/sn_testnet-v0.3.18

Commit:
docs: new wallet client

actions-user committed:
chore(release): sn_cli-v0.88.20/sn_transfers-v0.14.38/sn_faucet-v0.2.20/sn_node-v0.102.20/sn_node_rpc_client-v0.2.20/sn_client-v0.101.12/sn_networking-v0.12.30/sn_protocol-v0.10.9

Commit:
docs: new wallet client example

JasonPaulGithub committed:
Merge branch 'main' into walletclientnew
